src/ray/core_worker/core_worker.h

```diff
@@ -1409,7 +1409,8 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
                       ObjectID object_id, void *py_future);
 
   /// Whether we are shutting down and not running further tasks.
-  bool exiting_ = false;
+  /// Being accessed by grpc thread and task_execution_service_ thread.
+  std::atomic<bool> exiting_;
```

Comment:
Good to comment on the behavior.

Btw, why not keep setting its default value?

Does it matter to call `exiting_.load(std::memory_order_relaxed)` when it is read?